include/unifex/sequence.hpp

```diff
@@ -0,0 +1,493 @@
+            Successor,
+            sequence_successor_receiver<Predecessor, Successor, Receiver>>>
+            succOp_;
+      };
+    };
+  }  // namespace detail
+
+  template <typename Predecessor, typename Successor>
+  class sequence_sender {
+  public:
+    template <
+        template <typename...>
+        class Variant,
+        template <typename...>
+        class Tuple>
+    using value_types =
+        typename Successor::template value_types<Variant, Tuple>;
+
+    template <template <typename...> class Variant>
+    using error_types = concat_unique_t<
+        typename Predecessor::template error_types<Variant>,
+        typename Successor::template error_types<Variant>>;
+
+    template <
+        typename Predecessor2,
+        typename Successor2,
+        std::enable_if_t<
+            std::is_constructible_v<Predecessor, Predecessor2> &&
+                std::is_constructible_v<Successor, Successor2>,
+            int> = 0>
+    explicit sequence_sender(
+        Predecessor2&& predecessor,
+        Successor2&&
+            successor) noexcept(std::
+                                    is_nothrow_constructible_v<
+                                        Predecessor,
+                                        Predecessor2>&&
+                                        std::is_nothrow_constructible_v<
+                                            Successor,
+                                            Successor2>)
+      : predecessor_(static_cast<Predecessor&&>(predecessor))
+      , successor_(static_cast<Successor&&>(successor)) {}
+
+    friend blocking_kind
+    tag_invoke(tag_t<blocking>, const sequence_sender& sender) {
+      const auto predBlocking = blocking(sender.predecessor_);
+      const auto succBlocking = blocking(sender.successor_);
+      if (predBlocking == blocking_kind::never) {
+        return blocking_kind::never;
+      } else if (
+          predBlocking == blocking_kind::always_inline &&
+          succBlocking == blocking_kind::always_inline) {
+        return blocking_kind::always_inline;
+      } else if (
+          (predBlocking == blocking_kind::always_inline ||
+           predBlocking == blocking_kind::always) &&
+          (succBlocking == blocking_kind::always_inline ||
+           succBlocking == blocking_kind::always)) {
+        return blocking_kind::always;
+      } else {
+        return blocking_kind::maybe;
+      }
+    }
+
+    template <
+        typename Receiver,
+        std::enable_if_t<
+            is_connectable_v<
+                Predecessor,
+                detail::sequence_predecessor_receiver<
+                    Predecessor,
+                    Successor,
+                    std::remove_cvref_t<Receiver>>> &&
+                is_connectable_v<
+                    Successor,
+                    detail::sequence_successor_receiver<
+                        Predecessor,
+                        Successor,
+                        Receiver>>,
+            int> = 0>
```

Comment:
Say what you will, but IMO the concepts emulation macros from the range library would be an improvement here.